### PR TITLE
Rewrote how to choose a getting-started-guide.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@thockin @bgrant0607 @RichieEscarez @davidopp 

I rewrote docs/getting-started-guide/README.md.  It still needs a pass of proofreading, but I thought I would get the general format out there for feedback.

I've switched from having a table as the first thing you see to having prose that helps guide your decision of where to start.  I left the table in at the bottom, but I'm thinking of getting rid of it too in a follow-up PR.

I've started replacing "getting started guide" with the word "solution".  Here is why:
  - "getting started guide" is a mouthful.
  - I don't want to imply that once you created a cluster, you have gotten started.  getting started means really using kubectl to create some pods. 

My thought in terms of a left-nav for kubernetes.io is: to have the following items somewhere on the left nav bar:
- *Creating a cluster* links to docs/getting-started-guides/README.md
  - *Creating a local cluster* links to docs/getting-started-guides/README.md#local-solutions
  - *Creating a hosted cluster* links to docs/getting-started-guides/README.md#hosted-solutions
  - *Creating a cluster in the cloud* links to docs/getting-started-guides/README.md#turn-key-cloud-solutions
  - *Creating a custom cluster* links to docs/getting-started-guides/README.md#custom-solutions

**Comment by bgrant0607** [comment]
Awesome! Thanks much. I think this will help a lot.
LGTM

**Comment by bgrant0607** [comment]
cc @davidopp for merge

File: docs/getting-started-guides/README.md [hunk 1]
```diff
@@ -20,13 +20,99 @@ certainly want the docs that go with that version.</h1>
...
+
+[Turn-key cloud solutions](#turn-key-cloud-solutions) require only a few commands to create
+and cover a wider range of cloud providers.
+
+[Custom solutions](#custom-solutions) require more effort to setup but cover and even 
+they vary from step-by-step instructions to general advice for setting up
+a kubernetes cluster from scratch. 
+
+### Local-machine Solutions
+These local-machine solutions create a single cluster with one or more kubernetes nodes on a single
```
Comment (davidopp) [inline docs/getting-started-guides/README.md]: What does "These" refer to? Did you mean to point to a doc or docs here?

File: docs/getting-started-guides/README.md [hunk 2]
```diff
@@ -20,13 +20,99 @@ certainly want the docs that go with that version.</h1>
...
+
+### Hosted Solutions
+[Google Container Engine](https://cloud.google.com/container-engine) offers managed Kubernetes
+clusters.
+
+### Turn-key Cloud Solutions
+These solutions allow you to create Kubernetes clusters on range of Cloud IaaS providers with only a
+few commands, and have active community support.
+
+GCE                  | Saltstack    | Debian | GCE         | [docs](gce.md)                                    | [✓][1]   | Project
```
Comment (davidopp) [inline docs/getting-started-guides/README.md]: This chart did not render properly when I viewed it.

File: docs/getting-started-guides/README.md (hunk 1 above)
Comment (davidopp) [inline docs/getting-started-guides/README.md]: If you were referring to the ones listed in the next paragraph, then get rid of "Other automated local-machine solutions are:" and just move the bullet list up to merge with this paragraph

File: docs/getting-started-guides/README.md [hunk 3]
```diff
@@ -20,13 +20,99 @@ certainly want the docs that go with that version.</h1>
...
+
+Kubernetes can run on a wide range of Cloud providers and bare-metal environments, and with many
+base operating systems.
+
+If you can find a guide below that matches your needs, use it.  It may be a little out of date, but
+it will be easier than starting from scratch.  If you do want to start from scratch, because you
+have special requirements, or just because you want to understand what is underneath a Kubernetes
+cluster, try the [Getting Started from Scratch](scratch.md) guide.
+
+If you do are interested in supporting Kubernetes on a new platform, checkout out our [advice for
```
Comment (davidopp) [inline docs/getting-started-guides/README.md]: do are -> are
checkout -> check out

**Comment by davidopp** [comment]
I'll leave the ok-to-merge label but please fix the above

File: docs/getting-started-guides/README.md (hunk 1 above)
Comment (erictune) [inline docs/getting-started-guides/README.md]: fixed.

File: docs/getting-started-guides/README.md (hunk 2 above)
Comment (erictune) [inline docs/getting-started-guides/README.md]: fixed.

File: docs/getting-started-guides/README.md (hunk 3 above)
Comment (erictune) [inline docs/getting-started-guides/README.md]: fixed

**Comment by davidopp** [comment]
LGTM

will merge on green

File: docs/getting-started-guides/README.md [hunk 4]
```diff
@@ -20,13 +20,98 @@ certainly want the docs that go with that version.</h1>
...
+and cover a wider range of cloud providers.
+
+[Custom solutions](#custom-solutions) require more effort to setup but cover and even 
+they vary from step-by-step instructions to general advice for setting up
+a kubernetes cluster from scratch. 
+
+### Local-machine Solutions
+Local-machine solutions create a single cluster with one or more kubernetes nodes on a single
+physical machine.  Setup is completely automated and doesn't require a cloud provider account.
+But their size and availability is limited to that of a single machine.
```
Comment (RichieEscarez) [inline docs/getting-started-guides/README.md]: Change 'But' to "However, "

File: docs/getting-started-guides/README.md [hunk 5]
```diff
@@ -20,13 +20,98 @@ certainly want the docs that go with that version.</h1>
...
+  - [Vagrant](vagrant.md) (works on any platform with Vagrant: Linux, MacOS, or Windows.)
+  - [No-VM local cluster](locally.md) (Linux only)
+
+
+### Hosted Solutions
+[Google Container Engine](https://cloud.google.com/container-engine) offers managed Kubernetes
+clusters.
+
+### Turn-key Cloud Solutions
+These solutions allow you to create Kubernetes clusters on range of Cloud IaaS providers with only a
```
Comment (RichieEscarez) [inline docs/getting-started-guides/README.md]: typo: insert 'a' -> ...a range of...

File: docs/getting-started-guides/README.md [hunk 6]
```diff
@@ -20,13 +20,98 @@ certainly want the docs that go with that version.</h1>
...
-[guidelines](../../docs/devel/writing-a-getting-started-guide.md).
+Kubernetes can run on a range of platforms, from your laptop, to VMs on a cloud provider, to rack of
+bare metal servers.  The effort required to setup a cluster varies from running a single command to
+crafting your own customized cluster.  We'll guide you in picking a solution that fits for your needs.
+
+## Picking the Right Solution
+
+If you just want to "kick the tires" on Kubernetes, we recommend the [local Docker-based](docker.md) solution.  
+
+The local Docker-based solution is one of several [Local cluster](#local-solutions) solutions
```
Comment (RichieEscarez) [inline docs/getting-started-guides/README.md]: this anchor link is not working -> local-machine-solutions

File: docs/getting-started-guides/README.md [hunk 7]
```diff
@@ -20,13 +20,98 @@ certainly want the docs that go with that version.</h1>
...
+they vary from step-by-step instructions to general advice for setting up
+a kubernetes cluster from scratch. 
+
+### Local-machine Solutions
+Local-machine solutions create a single cluster with one or more kubernetes nodes on a single
+physical machine.  Setup is completely automated and doesn't require a cloud provider account.
+But their size and availability is limited to that of a single machine.
+
+The local-machine solutions are:
+  - [Local Docker-based](#local-solutions) (recommended starting point)
```
Comment (RichieEscarez) [inline docs/getting-started-guides/README.md]: this is the wrong link and is not working

File: docs/getting-started-guides/README.md [hunk 8]
```diff
@@ -20,13 +20,98 @@ certainly want the docs that go with that version.</h1>
...
+- [Fedora single node](fedora/fedora_manual_config.md) 
+- [Fedora multi node](fedora/flannel_multi_node_cluster.md) 
+- [Centos](centos/centos_manual_config.md)
+- [Ubuntu](ubuntu.md)
+- [Docker Multi Node](docker-multinode.md)
+
+#### Integrations
+- [Kubernetes on Mesos](mesos.md) (Uses GCE)
+
+## Table of Solutions
```
Comment (RichieEscarez) [inline docs/getting-started-guides/README.md]: Can we replace this with a link, move this "table of solutions" to its own document, and say something like:

"See all of the solutions in a table, ordered by last tested/supported product version."

**Comment by RichieEscarez** [comment]
Thanks Eric. Looks great! Ill update the site nav based on this structure. LGTM pending my comments.

**Pull request merged**